### ISSUE TITLE
Bold if same colors

### ISSUE THREAD

**Issue by lexazloy** [mintty/mintty#478]
If i using same color and lite color i have lite color as bold:
```
echo -ne "\e]4;1;#ffffff\a"
echo -ne "\e]4;9;#ffffff\a"

echo -e "Naked \e[1;31mpeople\e[0m dancing"
```

> Naked **people** dancing

But if colors not same work as expected:

```
echo -ne "\e]4;1;#ffffff\a"
echo -ne "\e]4;9;#fffffE\a" # here

echo -e "Naked \e[1;31mpeople\e[0m dancing"
```

> Naked people dancing

P.S. `Show bold as font` unchecked, `Show bold as colour` checked.

**Comment by mintty** [comment]
This is related to #468 but not quite the same issue.

You have redefined ANSI colours 1 and 9 and they do in fact behave as expected:
```
echo -ne "\e[38;5;1m ANSI colour 1 \e[0m"
echo -ne "\e[38;5;9m ANSI colour 9 \e[0m"
```
Both show the same redefined colour.

The question here is whether the ANSI bold attribute (\e[1m) should (with your config options) always simply map ANSI colours 0..7 to 8..15, overriding the principle which I'm trying to defend in #468 not to support "display bold as non-bold". What actually happens is that overstriking is applied as a fall-back to enforce distinct bold display, i.e. mintty is not as you said showing "lite as bold" but rather "bold as bold" - which is more or less what **I** would expect. However, it is arguable whether explicit redefining of the "colour implementing bold" should make an exception here. I will consider this and also check how other terminals behave.

**Comment by lexazloy** [comment]
Thank you for answer. I saw previous ticket but dont understand how similar (or not) issue is.

If you say it pretty expected situation then ok. I mean i will little change second color and it give what i expect. Not a big deal.

But reason why i start issue. I tested few terms and every show my "version of truth". :)

Here Xfce4 Terminal, Xterm and Mintty:
![](https://i.imgur.com/UMZ44Kn.png)

Lxterminal:
![](https://i.imgur.com/6Tnf4LM.png)

Gnome Terminal:
![](https://i.imgur.com/FC96iag.png)

**Comment by mintty** [comment]
Changing policy; with the latest change, overstriked bold display is not enforced anymore if the colour that the bold attribute gets mapped to is explicitly redefined (and happens to be the same as the non-bold colour).
This applies to the default foreground colour (mapped to the new bold colour which is configurable with option BoldColour or with the corresponding xterm control sequences), and the ANSI colours 0 to 7 (which are mapped to ANSI colours 8 to 15).

Default behaviour (i.e. without colour redefinitions) is still (and will remain to be) overstriking bold mode (if BoldAsFont=false).

**Comment by mintty** [comment]
Revised in 2.2.1.

**Comment by silverwind** [comment]
I see `BoldAsFont=no` and `BoldAsColour=yes` is now removing the overstrike from ANSI colors 0-7, but not from the "light" variants. What's kind of weird is that even non-bold light colors result in a bold rendering:

![](https://cloud.githubusercontent.com/assets/115237/20866673/a6370d22-ba32-11e6-9e72-51776b0764ae.png)

Is there a way I can stop these from getting bolded too? Here's how I define them in my shell startup file:

````
echo -ne '\e]10;#D1C092\a' # ForegroundColour
echo -ne '\e]11;#080808\a' # BackgroundColour

echo -ne '\e]12;#ffaf00\a' # CursorColour
echo -ne '\e]4;262;#80e0a0\a' # IMECursorColour

echo -ne '\e]4;0;#666666\a' # Black
echo -ne '\e]4;8;#999999\a' # LightBlack
echo -ne '\e]1;30;#666666\a' # Black

echo -ne '\e]4;1;#e52222\a' # Red
echo -ne '\e]4;9;#e52222\a' # LightRed

echo -ne '\e]4;2;#a6e32d\a' # Green
echo -ne '\e]4;10;#a6e32d\a' # LightGreen

echo -ne '\e]4;3;#fc951e\a' # Yellow
echo -ne '\e]4;11;#fc951e\a' # LightYellow

echo -ne '\e]4;4;#7575ff\a' # Blue
echo -ne '\e]4;12;#7575ff\a' # LightBlue

echo -ne '\e]4;5;#fa2573\a' # Magenta
echo -ne '\e]4;13;#fa2573\a' # LightMagenta

echo -ne '\e]4;6;#67d9f0\a' # Cyan
echo -ne '\e]4;14;#67d9f0\a' # LightCyan

echo -ne '\e]4;7;#E6DDB8\a' # White
echo -ne '\e]4;15;#E6DDB8\a' # LightWhite
````

**Comment by mintty** [comment]
Non-bold "bright" (rather than "light") colours (i.e. ANSI colours 8...15) are not displayed boldened; 
your test case is wrong, it's not resetting the bold attribute.

With the bold text attribute, BoldAsColour has an effect on the default foreground (using a calculated brightened colour or the configured BoldColour) and on the 8 ANSI colours 0...7 which are mapped to the bright colours 8...15. It does not have a colour effect on other colours (8...255 and true colour).

In general, there are 3 options to implement the bold attribute: BoldAsFont, BoldAsColour (but only for colours 0...7 and default as described above), and manual/overstrike boldening.
So if you disable BoldAsFont and use a colour other than default or 0...7, there is only one option left to actually display the bold attribute.
This assumes that you want to make text bold if you use the bold attribute which would otherwise not make much sense on colours 8...15 (which are already considered bold in a sense by their BoldAsColour mapping from 0...7).

Any further arguments will be considered but as stated before, I don't think "bold as non-bold" is a reasonable option. If you don't want bold, don't use the bold attribute in the first place.
The only thing I can imagine is to apply calculated brightening (like currently for the default foreground) also to colours 8...15. Not sure whether that's much benefit, considering that it doesn't appear useful to apply the same algorithm to palette colours 16...256 and true colour.

**Comment by mintty** [comment]
Released 2.9.1.
